### PR TITLE
[BD-38] [TNL-9266] [BB-5143] Add reason code selection UI for moderators on posts

### PR DESCRIPTION
**Pull request by felipetrz**
### Progress

[x] Add selector for edit reason codes on post and comment editing
[x] Add indicator in the posts list that the post was edited by a moderator
[ ] Add tests

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @felipetrz! I've created [OSPR-6297](https://openedx.atlassian.net/browse/OSPR-6297) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: src/discussions/posts/data/api.js [hunk 1]
```diff
@@ -152,6 +154,7 @@ export async function updateThread(threadId, {
     following,
     closed,
     pinned,
+    edit_reason_code: editReasonCode,
```
Comment (xitij2000) [inline src/discussions/posts/data/api.js:157]: `snakeCaseObject` will take care of converting this. 
```suggestion
    editReasonCode,
```

File: src/discussions/posts/post-editor/PostEditor.jsx [hunk 2]
```diff
@@ -301,6 +307,25 @@ function PostEditor({
               value={values.title}
             />
             <FormikErrorFeedback name="title" />
+            {(editExisting && userIsPrivileged) && (
```
Comment (xitij2000) [inline src/discussions/posts/post-editor/PostEditor.jsx:310]: I think we should add another check here, which is that the current user isn't the author of the post. 

i.e. if someone is editing their own post this shouldn't show up.

**Comment by tecoholic** [comment]
:+1:

- [X] I tested this: Set the edx-platform to [corresponding branch](https://github.com/openedx/edx-platform/pull/29831) and configured necessary Waffle flags and used the UI to do moderation with specific reasons.
- [X] I read through the code
- [X] I checked for accessibility issues - Checked for use of `Intl` for UI strings 
- [ ] Includes documentation -

**Review by tecoholic — APPROVED** [review]
:+1:

- [X] I tested this: Set the edx-platform to [corresponding branch](https://github.com/openedx/edx-platform/pull/29831) and configured necessary Waffle flags and used the UI to do moderation with specific reasons.
- [X] I read through the code
- [X] I checked for accessibility issues - Checked for use of `Intl` for UI strings 
- [ ] Includes documentation - NA

**Review by mehaknasir — APPROVED** [review]
the PR looks good to me.

**Comment by mehaknasir** [comment]
@xitij2000 also I checked that the error shown as default CI is actually a failing test case in src/discussions/common/AlertBanner.test.jsx in this test suit, can you please rebase and try again if that fixes the issue? Thanks.

File: src/discussions/posts/post/messages.js [hunk 3]
```diff
@@ -75,6 +75,14 @@ const messages = defineMessages({
...
     description: 'Label shown on the badge indicating new comments/posts like "3 new"',
   },
+  editedBy: {
+    id: 'discussions.post.editedBy',
+    defaultMessage: 'Edited by',
+  },
+  reason: {
+    id: 'discussions.post.reason',
+    defaultMessage: 'Reason',
+  },
```
Comment (awais-ansari) [inline src/discussions/posts/post/messages.js:85]: Just descriptions are missing.

**Comment by edx-webhook** [comment]
@felipetrz 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-discussions&entry.867055334=https://github.com/openedx/frontend-app-discussions/pull/46&entry.1484655318=https://github.com/felipetrz&entry.752974735=2021-12-17+17:08&entry.1917517419=2022-03-17+12:41&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**